fab3/filters.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package fab3
+
+import (
+	"encoding/hex"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/hyperledger/fabric-chaincode-evm/fab3/types"
+)
+
+type filterEntry interface {
+	LastAccessTime() time.Time
+	Filter(s *ethService) ([]interface{}, error)
+}
+
+type logsFilter struct {
+	gla            *types.GetLogsArgs
+	lastAccessTime time.Time
+}
+
+func (f *logsFilter) LastAccessTime() time.Time {
+	return f.lastAccessTime
+}
+
+func (f *logsFilter) Filter(s *ethService) ([]interface{}, error) {
+	var logs []types.Log
+	err := s.GetLogs(nil, f.gla, &logs)
```

Comment:
Instead of calling GetLogs every time a filter is asked, i think we should be making use of the event stream that we can get from the sdk. We could cache the events that match a filter and clear that cache per filter when those filter changes are asked. If a filter expires we can remove it then and delete its associated events. 

I think using the event stream will reduce the number of requests we send the peer, especially repeat ones if dapps use multiple filters. It also means we don't need to keep track things like latestBlockSeen. 

I do understand that this implementation reuses a lot of what we have already built, but I think we should optimize on reducing requests to the peer.